core/src/main/java/org/frankframework/parameters/AbstractParameter.java

```diff
@@ -601,17 +613,10 @@ private Object getValueForFormatting(ParameterValueList alreadyResolvedParameter
 			if (substitutionValueMessage != null) {
 				if (substitutionValueMessage instanceof Date substitutionValueDate) {
 					substitutionValue = getSubstitutionValueForDate(substitutionValueDate, formatType);
+				} else if (substitutionValueMessage instanceof String stringValue) {
+					substitutionValue = preFormatDateType(stringValue, formatType, formatString);
 				} else {
-					if (substitutionValueMessage instanceof String stringValue) {
-						substitutionValue = preFormatDateType(stringValue, formatType, formatString);
-					} else {
-						try {
-							Message message = Message.asMessage(substitutionValueMessage); // Do not close object from session here; might be reused later
-							substitutionValue = message.asString();
-						} catch (IOException e) {
-							throw new ParameterException(getName(), "Cannot get substitution value from session key: " + name, e);
-						}
-					}
+					substitutionValue = session.getString(name);
```

Comment:
Nu verplaatsen we natuurlijk het probleem maarja, iig 1 occurance minder....

Reply:
We verplaatsen het probleem naar een plek waar we het al opgelost hadden, hopelijk. 😃 
En hoeven daardoor minder if/else/try/catch te doen in de Parameter code.